### ISSUE TITLE
3.0.2 breaks uploader behavior with rollback and deletes files in remote

### ISSUE THREAD

**Issue by rajyan** [carrierwaveuploader/carrierwave#2685]
## Description

From 3.0.2, carrierwave started to delete files in remote when rollback happens. (which causes a 404 link)

I haven't found the root cause yet, but confirmed that it doesn't happen in 3.0.1.

Still looking for the cause in
https://github.com/carrierwaveuploader/carrierwave/pull/2679

## sample code

```ruby
ActiveRecord::Base.transaction do
  vals.map do |val, i|
    attachment = post.attachments.find(val["id"])
    attachment.some_column = val["col"] # not the mounted column
    attachment.save!
  end
end
```

post model has a has_many association with attachment model, which mounts a uploader.
when the transaction is rolled back by `find` or `save!`, file is deleted for attachment that is saved in the previous iteration.

**Comment by rajyan** [comment]
I'm trying to create a failing test case right now.

**Comment by rajyan** [comment]
> I'm trying to create a failing test case right now.

Done.
This seems to be a severe regression in some use cases.

https://github.com/carrierwaveuploader/carrierwave/pull/2686

**Comment by rajyan** [comment]
@mshibuya 

Could you please cut a new release?
Although we have reverted to 3.0.1 and not affected by this issue now, I believe this issue seems quite critical, because it might delete users files in remote.

**Comment by mshibuya** [comment]
You're right, just released 3.0.3.

**Comment by rajyan** [comment]
Thank you!
